pkg/apis/apps/v1beta2/conversion.go

```diff
@@ -361,3 +368,89 @@ func Convert_extensions_RollingUpdateDeployment_To_v1beta2_RollingUpdateDeployme
 	}
 	return nil
 }
+
+func Convert_extensions_DaemonSet_To_v1beta2_DaemonSet(in *extensions.DaemonSet, out *appsv1beta2.DaemonSet, s conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	if out.Annotations == nil {
+		out.Annotations = make(map[string]string)
+	}
+	out.Annotations[appsv1beta2.DeprecatedTemplateGeneration] = strconv.FormatInt(in.Spec.TemplateGeneration, 10)
+	if err := Convert_extensions_DaemonSetSpec_To_v1beta2_DaemonSetSpec(&in.Spec, &out.Spec, s); err != nil {
+		return err
+	}
+	if err := s.Convert(&in.Status, &out.Status, 0); err != nil {
```

Comment:
Our roundtrip test doesn't exercise these conversions, because the internal Kind and the external Kind are of different group (see [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip/roundtrip.go#L210-L212)). We need to update `roundTripToAllExternalVersions` to test cross-group conversions. 

@sttts @deads2k @lavalamp do we have existing tests that exercise cross-group conversions?